### PR TITLE
Add serialization for ScriPlusInterpolationManager

### PR DESCRIPTION
**Pull request by moxcodes**
## Proposed changes

Adds a missing pup function to the `ScriPlusInterpolationManager`. Split out from #2374.

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by fmahebert — DISMISSED** [review]

File: src/Evolution/Systems/Cce/ScriPlusInterpolationManager.hpp [hunk 1]
```diff
@@ -153,8 +163,8 @@ struct ScriPlusInterpolationManager {
   std::deque<VectorTypeToInterpolate> to_interpolate_values_;
   std::deque<std::pair<double, double>> u_bondi_ranges_;
   std::deque<double> target_times_;
-  size_t vector_size_ = 0;
-  size_t target_number_of_points_ = 0;
+  size_t vector_size_ = 0_st;
```
Comment (nilsdeppe) [inline src/Evolution/Systems/Cce/ScriPlusInterpolationManager.hpp:169]: need include `Utilities/Literals.hpp`

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
you can squash :)

**Comment by moxcodes** [comment]
@nilsdeppe done.
Cheers!

File: src/Evolution/Systems/Cce/ScriPlusInterpolationManager.hpp [hunk 2]
```diff
@@ -143,6 +144,16 @@ struct ScriPlusInterpolationManager {
     return u_bondi_ranges_.size();
   }
 
+  void pup(PUP::er& p) noexcept {  // NOLINT
```
Comment (kidder) [inline src/Evolution/Systems/Cce/ScriPlusInterpolationManager.hpp:149]: also need to include the appropriate pup includes

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

**Comment by moxcodes** [comment]
@kidder done.
Cheers!

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**